anvil/src/shell.rs

```diff
@@ -480,26 +495,27 @@ impl<BackendData: Backend> XdgShellHandler for AnvilState<BackendData> {
 
         let start_data = pointer.grab_start_data().unwrap();
 
+        let window = self
+            .space
+            .elements()
+            .find(|window| window.toplevel().wl_surface() == surface.wl_surface())
+            .unwrap()
+            .clone();
```

Comment:
We seem to have a lot of these in Anvil. Maybe we could have a function to do this within anvil.